java/ql/src/experimental/Security/CWE/CWE-552/UnsafeUrlForward.ql

```diff
@@ -43,6 +44,20 @@ class UnsafeUrlForwardFlowConfig extends TaintTracking::Configuration {
   override DataFlow::FlowFeature getAFeature() {
     result instanceof DataFlow::FeatureHasSourceCallContext
   }
+
+  override predicate isAdditionalTaintStep(DataFlow::Node prev, DataFlow::Node succ) {
```

Comment:
These taint steps should probably have tests.

Reply:
I've added missing test cases for taint steps of class loader, JSF, and JBoss. Please review. Thanks.